[BUG] Build failure when spinlock debugging is on

**Describe the bug**
Enabling spinlock debugging ends up in a build failure.

**To Reproduce**
Steps to reproduce the behavior: (e.g. list commands or actions used to reproduce the bug)

     make menuconfig
     Enable Debug --->
     [*] Spinlock debug

The build fails with:

    [ 52%] Building C object CMakeFiles/sof.dir/src/arch/xtensa/schedule/schedule.c.o
     xtensa-byt-elf-gcc  -DRELATIVE_FILE=\"src/arch/xtensa/schedule/schedule.c\" -Isof/src/arch/xtensa/include -I/sof/src/arch/xtensa/xtos -Isof/src/platform/baytrail/include -Isof/src/platform/baytrail/include/arch -Isof/sof/../xtensa-root/xtensa-byt-elf/include -Isof/src/include -Isof/build_byt_gcc/generated/include  -nostdlib -fno-inline-functions -mlongcalls -O2 -g -Wall -Werror -Wl,-EL -Wmissing-prototypes -Wpointer-arith -mtext-section-literals -o CMakeFiles/sof.dir/src/arch/xtensa/schedule/schedule.c.o   -c sof/src/arch/xtensa/schedule/schedule.c
    In file included from sof/src/include/sof/trace/trace.h:23,
                 from sof/src/include/sof/schedule/schedule.h:16,
                 from sof/src/arch/xtensa/schedule/schedule.c:14:
    sof/src/include/sof/spinlock.h: In function '_spin_lock':
    sof/src/include/sof/spinlock.h:80:36: error: implicit declaration of function 'trace_error_atomic' [-Werror=implicit-function-declaration]

Attempted quick fix at https://github.com/thesofproject/sof/pull/2332 does not resolve the circular include dependencies.

--- a/src/include/sof/spinlock.h
+++ b/src/include/sof/spinlock.h
@@ -15,7 +15,6 @@
 
 #include <arch/spinlock.h>
 #include <sof/lib/memory.h>
-#include <sof/trace/trace.h>
 #include <config.h>
 #include <stdint.h>
 

--- a/src/include/sof/trace/trace.h
+++ b/src/include/sof/trace/trace.h
@@ -20,7 +20,6 @@
 #endif
 #include <sof/common.h>
 #include <sof/sof.h>
-#include <sof/spinlock.h>
 #include <sof/trace/preproc.h>
 #include <config.h>
 #include <stdint.h>
@@ -29,12 +28,7 @@
 #endif
 
 struct sof;
-
-struct trace {
-	uint32_t pos ;	/* trace position */
-	uint32_t enable;
-	spinlock_t lock; /* locking mechanism */
-};
+struct trace;
 
 /* bootloader trace values */
 #define TRACE_BOOT_LDR_ENTRY		0x100

--- a/src/trace/trace.c
+++ b/src/trace/trace.c
@@ -23,6 +23,12 @@
 #include <user/trace.h>
 #include <stdint.h>
 
+struct trace {
+	uint32_t pos ;	/* trace position */
+	uint32_t enable;
+	spinlock_t lock; /* locking mechanism */
+};
+
 /* calculates total message size, both header and payload in bytes */
 #define MESSAGE_SIZE(args_num)	\
 	(sizeof(struct log_entry_header) + args_num * sizeof(uint32_t))

